vsintegration/src/FSharp.Editor/Diagnostics/SimplifyNameDiagnosticAnalyzer.fs

```diff
@@ -26,6 +26,7 @@ type internal SimplifyNameDiagnosticAnalyzer() =
     let getProjectInfoManager (document: Document) = document.Project.Solution.Workspace.Services.GetService<FSharpCheckerWorkspaceService>().ProjectInfoManager
     let getChecker (document: Document) = document.Project.Solution.Workspace.Services.GetService<FSharpCheckerWorkspaceService>().Checker
     let getPlidLength (plid: string list) = (plid |> List.sumBy String.length) + plid.Length
+    static let cache = ConditionalWeakTable<DocumentId, TextVersionHash * ImmutableArray<Diagnostic>>()
```

Comment:
>  automatically removes the key/value entry as soon as no other references to a key exist outside the table

That's nice, didn't know about that class, thanks for the pointer. So I guess the answer is it doesn't get deleted when document is closed but when it is garbage collected.